src/stats/src/StatisticalInverseProblem.C

```diff
@@ -141,14 +136,14 @@ StatisticalInverseProblem<P_V,P_M>::StatisticalInverseProblem(
 
   // If NULL, we create one
   if (m_optionsObj == NULL) {
-    SipOptionsValues * tempOptions = new SipOptionsValues(&m_env, prefix);
-
-    // We did this dance because scanOptionsValues is not a const method, but
-    // m_optionsObj is a pointer to const
-    m_optionsObj = tempOptions;
-
-    m_userDidNotProvideOptions = true;
+    m_optionsObj.reset(new SipOptionsValues(&m_env, prefix));
   }
+  else {
+    // If the user passed in a legitimate options object, then copy it into a
+    // ScopedPtr
+    m_optionsObj.reset(new SipOptionsValues(*alternativeOptionsValues));
```

Comment:
I agree that this is the right way to do things in this case, but we should document that we're keeping a copy rather than a reference.  Users are going to ask "Can I deconstruct my options object or does it need to outlive the SIP", and users are going to ask "Can I change my options object after the SIP is constructed", and there's no good way to indicate "Yes and no respectively" in code rather than docs.